TheraLang.Tests/Services/DonationServiceTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using Common.Enums;
+using Common.Exceptions;
+using FluentAssertions;
+using Moq;
+using TheraLang.BLL.DataTransferObjects.Donations;
+using TheraLang.BLL.LiqPay;
+using TheraLang.BLL.Services;
+using TheraLang.DAL.Entities;
+using TheraLang.DAL.Repository;
+using TheraLang.DAL.UnitOfWork;
+using Xunit;
+
+namespace TheraLang.Tests.Services
+{
+    public class DonationServiceTests
+    {
+        private readonly Mock<IUnitOfWork> _unitOfWorkMock;
+        private readonly Mock<ILiqPayInfo> _liqPayInfoMock;
+        private readonly DonationService _donationService;
+
+        private readonly List<Donation> _testDb;
+        private readonly LiqPayCheckoutDto _validCheckout;
+
+        public DonationServiceTests()
+        {
+            _testDb = GetTestData().ToList();
+            _validCheckout = new LiqPayCheckoutDto()
+            {
+                Data = "eyJhY3Rpb24iOiJwYXkiLCJwYXltZW50X2lkIjoxMjUzMTU2NzE2LCJzdGF0dXMiOiJzdWNjZXNzIiwidmVyc2lvbiI6MywidHlwZSI6ImJ1eSIsInBheXR5cGUiOiJjYXJkIiwicHVibGljX2tleSI6InNhbmRib3hfaTQyODU5OTk4OTE0IiwiYWNxX2lkIjo0MTQ5NjMsIm9yZGVyX2lkIjoiSElVQjY1QlQxNTgyNTU1MTQyMTYyMDY1IiwibGlxcGF5X29yZGVyX2lkIjoiMEpYTDZVU1ExNTgyNTU1MTUwMjgyNjM2IiwiZGVzY3JpcHRpb24iOiJibGFnb2RpeW5pc3QiLCJzZW5kZXJfY2FyZF9tYXNrMiI6IjQyNDI0Mio0MiIsInNlbmRlcl9jYXJkX2JhbmsiOiJUZXN0Iiwic2VuZGVyX2NhcmRfdHlwZSI6InZpc2EiLCJzZW5kZXJfY2FyZF9jb3VudHJ5Ijo4MDQsImlwIjoiMTk1LjE2MC4yMzIuMjQ4IiwiYW1vdW50IjoxMDAuMCwiY3VycmVuY3kiOiJVU0QiLCJzZW5kZXJfY29tbWlzc2lvbiI6MC4wLCJyZWNlaXZlcl9jb21taXNzaW9uIjoyLjc1LCJhZ2VudF9jb21taXNzaW9uIjowLjAsImFtb3VudF9kZWJpdCI6MjQ1MC45OCwiYW1vdW50X2NyZWRpdCI6MjQ1MC45OCwiY29tbWlzc2lvbl9kZWJpdCI6MC4wLCJjb21taXNzaW9uX2NyZWRpdCI6NjcuNCwiY3VycmVuY3lfZGViaXQiOiJVQUgiLCJjdXJyZW5jeV9jcmVkaXQiOiJVQUgiLCJzZW5kZXJfYm9udXMiOjAuMCwiYW1vdW50X2JvbnVzIjowLjAsIm1waV9lY2kiOiI3IiwiaXNfM2RzIjpmYWxzZSwibGFuZ3VhZ2UiOiJ1ayIsImNyZWF0ZV9kYXRlIjoxNTgyNTU1MTUwMjg0LCJlbmRfZGF0ZSI6MTU4MjU1NTE1MDg3MSwidHJhbnNhY3Rpb25faWQiOjEyNTMxNTY3MTZ9",
+                Signature = "unholN06qwKFFl/LB3tc1qegq4E="
+            };
+            
+            _liqPayInfoMock = new Mock<ILiqPayInfo>();
+            _liqPayInfoMock.Setup(l => l.PrivateKey).Returns("testPrivateKey");
+            _liqPayInfoMock.Setup(l => l.PublicKey).Returns("testPublicKey");
+
+            var repoMock = new Mock<IRepository<Donation>>();
+            repoMock.Setup(r => r.Get(It.IsAny<Expression<Func<Donation, bool>>>()))
```

Comment:
do we know what values should be here? if so any we use no Any but the specific arugment?

Reply:
This setup we could use in multiple tests
e.g passing valid/invalid expression, null